packages/client/modules/meeting/components/MeetingAvatarGroup/NewMeetingAvatarGroup.tsx

```diff
@@ -75,62 +76,45 @@ interface Props {
   allowVideo: boolean
 }
 
+const MAX_AVATARS_DESKTOP = 7
+const MAX_AVATARS_MOBILE = 3
 const NewMeetingAvatarGroup = (props: Props) => {
   const atmosphere = useAtmosphere()
+  const {viewerId} = atmosphere
   const {swarm, team, camStreams, allowVideo} = props
   const {teamMembers} = team
-  const isOverflowBreakpoint = useBreakpoint(Breakpoint.SINGLE_REFLECTION_COLUMN)
+  const isDesktop = useBreakpoint(Breakpoint.SINGLE_REFLECTION_COLUMN)
   // all connected teamMembers except self
   // TODO: filter by team members who are actually viewing “this” meeting view
-  const connectedTeamMembers =
-    teamMembers &&
-    useMemo(() => teamMembers.filter(({isSelf, isConnected}) => !isSelf && isConnected), [
-      teamMembers
-    ])
-  const self = teamMembers && useMemo(() => teamMembers.find(({isSelf}) => isSelf), [teamMembers])
-  // on mobile, show self and 2 other avatars, or self + count
-  const overflowMobileThreshold = 3
-  const showOverflowMobile =
-    !isOverflowBreakpoint && connectedTeamMembers.length > overflowMobileThreshold
-  // on laptop+, show self and 7 other avatars, or self and 6 other avatars + count
-  const overflowLaptopThreshold = 7
-  const showOverflowLaptop =
-    isOverflowBreakpoint && connectedTeamMembers.length > overflowLaptopThreshold
-  const overflowCount =
-    (showOverflowMobile && connectedTeamMembers.length) ||
-    (showOverflowLaptop && connectedTeamMembers.length - overflowLaptopThreshold)
-  // on laptop+, slice off the first few to show if past threshold
-  const otherTeamMembers = showOverflowLaptop
-    ? connectedTeamMembers.slice(0, overflowLaptopThreshold - 1)
-    : connectedTeamMembers
+  const connectedTeamMembers = useMemo(() => {
+    return teamMembers
+      .filter(({user}) => user.isConnected)
+      .sort((a, b) => (a.userId === viewerId ? -1 : a.checkInOrder < b.checkInOrder ? -1 : 1))
+  }, [teamMembers])
+  const overflowThreshold = isDesktop ? MAX_AVATARS_DESKTOP : MAX_AVATARS_MOBILE
+  const visibleConnectedTeamMembers = connectedTeamMembers.slice(0, overflowThreshold)
+  const hiddenTeamMemberCount = connectedTeamMembers.length - visibleConnectedTeamMembers.length
   return (
     <MeetingAvatarGroupRoot>
       <VideoControls
         allowVideo={allowVideo}
         swarm={swarm}
         localStreamUI={camStreams[atmosphere.viewerId]}
       />
-      {self && (
```

Comment:
@mattkrick I like what you’re doing here. One thing, the intent was to always at least show the local user’s avatar, not just the 1st of the connected team members via check-in sort order.

Reply:
I guess you’re checking that with `a.userId === viewerId ?` on 92? Testing locally now…

Reply:
yep! the tricky part is that the viewer may not be connected & we don't want to show them as connected if they aren't.

i guess we should design out what it should look like when the viewer isn't connected? maybe...
- don't show their avatar
- show others' avatars as faded
- show the overflow count avatar as gray